Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -590,6 +594,51 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
         }
 
         /// <summary>
+        /// <para>Check if there is damanged leftover data</para>
+        /// <para>If there is leftover data, will try to cleanup.</para>
+        /// <para>If fail to cleanup, will move leftover data to Temp folder</para>
+        /// </summary>
+        private static void EnsureInstallationEnviroment(string installationDir, ITracer tracer)
+        {
+            // folder is there but nupkg is gone, means previous uninstallation must encounter some error
+            bool isInstalledPackageBroken = FileSystemHelpers.DirectoryExists(installationDir) && FileSystemHelpers.GetFiles(installationDir, "*.nupkg").Length == 0;
+            if (!isInstalledPackageBroken)
+            {
+                return;
+            }
+
+            using (tracer.Step("There was leftover data from previous uninstallation. Trying to cleanup now."))
+            {
+                bool canCleanup = true;
+                try
+                {
+                    OperationManager.Attempt(() => FileSystemHelpers.DeleteDirectoryContentsSafe(installationDir, ignoreErrors: false));
```

Comment:
we can simply return if this thing don't throw, no?  No need to have canCleanup bool.